### ISSUE TITLE
Ticking World crash | Activation Range | Fabric 1.20.1

### ISSUE THREAD

**Issue by VOrlando520** [Wesley1808/ServerCore#112]
**Describe the bug**
A clear description of what the bug is.
Turn on Activation range, let it sit and randomly every few days the server crashes because of it. Here is the crash report: https://pastebin.com/DqfukZVV

**Reproduce**
Listed above

**Expected behavior**
A clear description of what you expected to happen.
Not to crash

**Versions** -Latest is not a version!
ServerCore: ServerCore- fabric-1.5.1-1.20.1
Minecraft: Fabric 1.20.1

**Mod incompatibilities**
List any installed mods you think might play a role in causing this bug.

**Comment by Wesley1808** [comment]
This doesn't seem right. All servercore does here to cause this crash is a simple vanilla lookup for entities in a bounding box on the server thread. The actual error happens purely in vanilla code, and the below state should not even really be possible unless another mod is doing something really cursed
```
java.lang.NullPointerException: Cannot invoke "net.minecraft.world.entity.EntityTrackingStatus.shouldTrack()" because the return value of "net.minecraft.world.entity.EntityTrackingSection.getStatus()" is null
```

I can also tell you right away I can't reproduce it by only just activating it as I've been running the mod in prod with activation range for years, so I'm going to need more info for this.

Do you have any other mods that might conflict (and could you attach the full crash report, not just the stacktrace?)

**Comment by VOrlando520** [comment]
[crash-2024-08-04_07.23.49-server.txt](https://github.com/user-attachments/files/16497037/crash-2024-08-04_07.23.49-server.txt)
Most of your standard optimization mods that im guessing you run as well. 

And when I say activating it and waiting, it isnt something that happens every day. Its for sporadic and happens randomly. Not set pattern just this is the crash almost every time.

**Comment by Wesley1808** [comment]
Yeah, I am pretty familiar with that kind of crash (which also makes it practically impossible for me to reproduce)
There's also a lot of other mods in there that I'm not too familiar with though, so I can't really say a lot about them.

I'm guessing it doesn't happen if you were to go back to the same area where that crash happened?
(`minecraft:resource_nether -6425 107 2280`)

Either way pretty weird crash, your best bet to find the cause would probably be a debug mod to detect when an entity section gets created or changed to a null status (and print a stacktrace to log the cause) because as far as I can tell from vanilla code that *should* be impossible.

**Comment by VOrlando520** [comment]
Yeah no crash when i go back to those coords. Is there any certain debug mod to use?

**Comment by Wesley1808** [comment]
I highly doubt it given how specific this issue is - you'll have to make it yourself or ask someone else to make it (I perhaps could but currently I'm a bit busy).
